sql: add barebones live cluster checks for creating DB regions

otan opened the pull request:
This can be moved at any point, but the live cluster checks seem
helpful. This is not yet persisted yet anywhere because we have not
decided to do this yet.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/56131)
<!-- Reviewable:end -->

arulajmani reviewed (changes requested):
One suggestion, but otherwise looks good. 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56131#-:-ML8THMcBjc0XsdIVDqx:b-ch1543)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani and @otan)

___
*[pkg/sql/descriptor.go, line 95 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/56131#-ML8T-5V7flYlwEKfn5G:-ML8T-5V7flYlwEKfn5H:b-88wl31) ([raw file](https://github.com/cockroachdb/cockroach/blob/97c3eb64dcf1478ea8d9edfca476a3ba09b42695/pkg/sql/descriptor.go#L95)):*
> ```Go
> 			return nil, false, err
> 		}
> 		regions := make([]string, 0, len(database.Regions))
> ```

Seems like this is just being used to check if there are duplicate regions -- can we make this a map so that we don't need to do the O(regions^2) thing below? 


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56131)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @arulajmani)

___
*[pkg/sql/descriptor.go, line 95 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/56131#-ML8T-5V7flYlwEKfn5G:-ML9YDv86hvTq1Ma4erV:bsp45e9) ([raw file](https://github.com/cockroachdb/cockroach/blob/97c3eb64dcf1478ea8d9edfca476a3ba09b42695/pkg/sql/descriptor.go#L95)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Seems like this is just being used to check if there are duplicate regions -- can we make this a map so that we don't need to do the O(regions^2) thing below? 
</blockquote></details>

was thinking that this would be rare such that it isn't worth creating the extra map, but happy to go either way.


<!-- Sent from Reviewable.io -->

otan committed:
sql: add barebones live cluster checks for creating DB regions

This can be moved at any point, but the live cluster checks seem
helpful. This is not yet persisted yet anywhere because we have not
decided to do this yet.

Release note: None

arulajmani reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/56131#-:-MLDbF5o2aIDEEEMqmrD:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan commented:
bors r=arulajmani
